kube-runtime/src/watcher.rs

```diff
@@ -475,9 +470,11 @@ where
 {
     match state {
         State::Empty => match wc.initial_list_strategy {
-            InitialListStrategy::ListWatch => {
-                (Some(Ok(Event::Init)), State::InitPage { continue_token: None })
-            }
+            InitialListStrategy::ListWatch => (Some(Ok(Event::Init)), State::InitPage {
+                continue_token: None,
+                objects: VecDeque::default(),
```

Comment:
If we know the page size in advance (since it's part of the strategy) would it make sense to allocate a ring buffer with a predetermined capacity equal to the page size? Bit of a micro optimisation I suppose 🤔 

**Edit**: nvm, this wouldn't work since we always construct a new `InitPage` for each page we consume. Would overcomplicate things.

Reply:
maybe there is something we can do here, but not sure what. it's also going to become "the old way" of doing things once this thing becomes stabilised and rolled out everywhere.